src/adapter/src/coord/sequencer.rs

```diff
@@ -587,6 +599,12 @@ impl Coordinator {
 
                     self.maybe_create_linked_cluster(source_id).await;
 
+                    // Progress subsources have their read policies handled
+                    // differently than all other sources and should not be
+                    // automatically compacted; instead they trail their
+                    // source's compaction commands.
+                    let initialize_compaction_read_policy = data_source != DataSource::Progress;
```

Comment:
Aljoscha opened up #17603, which will provide the fix for this API. I hadn't considered introducing another read capability to hold back the progress collection's `since`, which is far more natural than what I tried to do here.

Reply:
This was resolved in https://github.com/MaterializeInc/materialize/pull/17603